File: classes/class01/oscillator.py
import math

nu = 50 # Hz
omega = 2*math.pi # radians
t0 = 1.25 # 2

dt = 0.001 # Hz

A = 10 # cm
B = 2 # cm

for i in range(5000): # does not give 5000, only up to 4999
    t = i*dt
    x = (A*math.cos(omega*t) + B*math.cos(2*omega*t))*math.exp(-t/t0)
    print("{:10.4f} {:10.4f}".format(t,x))

File: classes/class03/oscillator.cpp
#include <cmath>
using std::cos;
using std::exp;

#include <iostream>
using std::cout;
using std::endl;

#include <iomanip>
using std::setw;

const double nu = 2.5; // Hz
const double omega = 2*M_PI*nu; // radians/s
const double t0 = 1.25; // S
const double dt = 0.001; // S
const double A = 10; // cm
const double B = 2; // cm


int main()
{
  for (int i = 0; i < 5000; ++i)
  {
    const double t = i*dt;
    const double x = (A*cos(omega*t) + B*cos(2*omega*t))*exp(-t/t0);
    cout << setw(20) << t << setw(20) << x << endl;
  }
  return 0;
}

 
